libfuse/dir.go

```diff
@@ -52,20 +53,21 @@ type Folder struct {
 	updateChan chan<- struct{}
 }
 
-func newFolder(fl *FolderList, h *libkbfs.TlfHandle) *Folder {
+func newFolder(fl *FolderList, h *libkbfs.TlfHandle, hname string) *Folder {
 	f := &Folder{
-		fs:    fl.fs,
-		list:  fl,
-		h:     h,
-		nodes: map[libkbfs.NodeID]fs.Node{},
+		fs:             fl.fs,
+		list:           fl,
+		h:              h,
+		hPreferredName: hname,
+		nodes:          map[libkbfs.NodeID]fs.Node{},
 	}
 	return f
 }
 
 func (f *Folder) name() libkbfs.CanonicalTlfName {
 	f.handleMu.RLock()
 	defer f.handleMu.RUnlock()
-	return f.h.GetCanonicalName()
+	return libkbfs.CanonicalTlfName(f.hPreferredName)
```

Comment:
probably better to have `name()` not return `CanonicalTlfName` now, since it can be different. Perhaps define a new type, say, `PreferredTlfName`, in `libkbfs` and use that? Make the various functions that return it return that type instead of string, etc.